### PR TITLE
Add support for custom arrow as function and per step offsets

### PR DESCRIPTION
**Pull request by erykpiast**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -46,41 +54,57 @@ export default class ReactUserTour extends Component {
...
-					offsetHeight: el.offsetHeight
+					offsetHeight: el.offsetHeight,
+					margin,
 				});
 			}
 			else {
-				elPos = positions.right({position});
+				elPos = positions.right({
+					position,
+					margin,
```
Comment (conorhastings) [inline src/index.js]: can you remove the trailing commas

File: src/index.js [hunk 2]
```diff
@@ -98,12 +122,26 @@ export default class ReactUserTour extends Component {
...
 			this.props.style.height,
-			currentTourStep.position
+			currentTourStep.position,
+			currentTourStep.margin,
+			currentTourStep.horizontalOffset,
+			currentTourStep.verticalOffset
 		);
 		const style = {...this.props.style};
+		const useCustomArrow = (customArrow) =>
+			"function" === typeof customArrow ?
```
Comment (conorhastings) [inline src/index.js]: nitpicky but can you reverse the order here ```typeof customArrow === "function"

File: src/index.js [hunk 3]
```diff
@@ -98,12 +122,26 @@ export default class ReactUserTour extends Component {
...
 			this.props.style.width,
 			this.props.style.height,
-			currentTourStep.position
+			currentTourStep.position,
+			currentTourStep.margin,
+			currentTourStep.horizontalOffset,
+			currentTourStep.verticalOffset
 		);
 		const style = {...this.props.style};
+		const useCustomArrow = (customArrow) =>
```
Comment (conorhastings) [inline src/index.js]: can this function be defined outside of render, better for performance if we don't have to define it every time we call render.

Comment (erykpiast) [inline src/index.js]: makes sense!

File: src/index.js (hunk 2 above)
Comment (erykpiast) [inline src/index.js]: as you wish.

File: test/index.js [hunk 4]
```diff
@@ -122,6 +122,24 @@ test("should not have a default arrow that will point to selected dom class if a
...
+test("should call render custom react component provided as arrow property", (assert) => {
+	const component = <Tour {...props} step={3} arrow={<span className="my-custom-arrow"/>}/>
+	const result = renderComponent(component);
+	assert.throws(() => throwNoClass(result, "react-user-tour-arrow"));
+	assert.ok(TestUtils.findRenderedDOMComponentWithClass(result, "my-custom-arrow"), "arrow exists");
+	ReactDOM.unmountComponentAtNode(node);
+	assert.end();
+});
+
+test("should call function provided as arrow property and render element returned by it", (assert) => {
```
Comment (conorhastings) [inline test/index.js]: 👍 thanks for adding tests

Comment (erykpiast) [inline test/index.js]: :)

**Comment by conorhastings** [comment]
@erykpiast published as `2.3.0` thanks for the contribution!

**Pull request merged**